### ISSUE TITLE
Embed protobuf targets in Bazel

### ISSUE THREAD

**Issue by abergmeier-dsfishlabs** [bazelbuild/bazel#3155]
### Description of the problem / feature request / question:

For people to be able to implement workers, it would be good if targets like `@io_bazel//src/main/protobuf:worker_protocol_cc_proto` and `@io_bazel//third_party/protobuf:protobuf_python` would be embedded into Bazel.
Possibly as targets `@io_bazel_worker//:cc_proto` and `@io_bazel_worker//:py_proto`.

**Comment by alexeagle** [comment]
These targets do exist, eg
`bazel build @bazel_tools//src/main/protobuf:worker_protocol.proto`

however, there's a broken reference (at least in Bazel 0.5.1)
```
ERROR: error loading package '@bazel_tools//src/main/protobuf': Extension file not found. Unable to load package for '@bazel_tools//third_party/protobuf/3.2.0:protobuf.bzl': BUILD file not found on package path.
```

Looks like `src/main/protobuf/BUILD` contains `load("//third_party/protobuf/3.2.0:protobuf.bzl", "cc_proto_library", "py_proto_library")` but there is no protobuf directory under third_party.

**Comment by alexeagle** [comment]
Same problem prevents implementing extra actions
https://docs.bazel.build/versions/master/be/extra-actions.html

```
alexeagle-macbookpro:rules_typescript alexeagle$ bazel info release
release 0.5.4-homebrew
alexeagle-macbookpro:rules_typescript alexeagle$ bazel build @bazel_tools//src/main/protobuf:extra_actions_base.proto
ERROR: error loading package '@bazel_tools//src/main/protobuf': Extension file not found. Unable to load file '@bazel_tools//tools/build_rules:genproto.bzl': file doesn't exist or isn't a file.
INFO: Elapsed time: 0.064s
```

**Comment by ulfjack** [comment]
I am definitely against bundling any additional things into the bazel binary.

I could maybe be convinced to add another git repo that only contains a copy of the proto (maintained by us) so that you can declare it as a workspace dep. On the other hand, I think it's also fine if you copy the proto to where you need it.

**Comment by ulfjack** [comment]
Let me rephrase that: I'd prefer not to bundle more things into the Bazel binary. Protos are forward and backwards-compatible (as long as we only make certain kinds of changes), so it shouldn't be necessary to use exactly the same proto as Bazel is using.

**Comment by aiuto** [comment]
Closing this as will not implement.
@ulfjack\'s comments about adding more to the bazel binary from 2 years ago still stand. Since then, we have made a concerted effort to reduce size, and to reduce the number of bundled tools.
A bazelbuild/bazel-lib repo might be a good place for just the .proto files and rules around them. Maybe we could synthesize one as part of the bazel release build. But let's start fresh on that idea if there really is demand.
